### PR TITLE
Bach unittest, convert tests to multi-database

### PR DESCRIPTION
**Pull request by thijs-obj**
Changes to bach functional tests:
* Replaced calls to `get_bt_with_test_data()` with calls to `get_df_with_test_data()`, where possible.
* Replaced calls to `get_bt_with_railway_data()` with calls to `get_df_with_railway_data()` ,where possible.
* Got rid of `get_bt_with_food_data()` and replaced calls to it with calls to `get_df_with_food_data()`.

### CODE REVIEW

File: bach/tests/functional/bach/test_df_setitem.py [hunk 1]
```diff
@@ -155,12 +157,16 @@ def test_set_series_column():
     )
     assert bt.duplicated_column == bt['duplicated_column']
 
-    bt['spaces in column'] = bt['founding']
+    spaced_col_name = 'spaces in column'
+    if is_bigquery(engine):
+        # BigQuery doesn't support spaces in column names. So we don't test that functionality on that engine
+        spaced_col_name = 'spaces_in_column'
```
Comment (KathiaBarahona) [inline bach/tests/functional/bach/test_df_setitem.py:163]: **question** isn't it better to just do a separate test for this and skip for bq?

**Review by KathiaBarahona — APPROVED** [review]
lgtm, just one small question. Nice we are getting rid of those functions

File: bach/tests/functional/bach/test_df_setitem.py (hunk 1 above)
Comment (thijs-obj) [reply bach/tests/functional/bach/test_df_setitem.py]: I think you are right. Fixed in 3dd06e59830cba9cd4aacd4f935a5c0c465e0567

**Comment by thijs-obj** [comment]
Discussed over chat. Was ok to merge after last change

**Pull request merged**